apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -190,3 +192,123 @@ impl<C: AsClient> crud::Read<Entity> for PostgresStore<C> {
         Ok(stream)
     }
 }
+
+#[derive(Debug)]
+pub struct EntityEdgeTraversalData {
+    owned_by_ids: Vec<OwnedById>,
+    entity_uuids: Vec<EntityUuid>,
+    entity_revision_ids: Vec<Timestamp<VariableAxis>>,
+    intervals: Vec<RightBoundedTemporalInterval<VariableAxis>>,
+    resolve_depths: Vec<GraphResolveDepths>,
+    pinned_timestamp: Timestamp<PinnedAxis>,
+    variable_axis: TimeAxis,
+}
+
+impl EntityEdgeTraversalData {
+    pub const fn new(pinned_timestamp: Timestamp<PinnedAxis>, variable_axis: TimeAxis) -> Self {
+        Self {
+            owned_by_ids: Vec::new(),
+            entity_uuids: Vec::new(),
+            entity_revision_ids: Vec::new(),
+            intervals: Vec::new(),
+            resolve_depths: Vec::new(),
+            pinned_timestamp,
+            variable_axis,
+        }
+    }
+
+    pub fn push(
+        &mut self,
+        vertex_id: EntityVertexId,
+        interval: RightBoundedTemporalInterval<VariableAxis>,
+        resolve_depth: GraphResolveDepths,
+    ) {
+        self.owned_by_ids.push(vertex_id.base_id.owned_by_id);
+        self.entity_uuids.push(vertex_id.base_id.entity_uuid);
+        self.entity_revision_ids.push(vertex_id.revision_id);
+        self.intervals.push(interval);
+        self.resolve_depths.push(resolve_depth);
+    }
+}
+
+pub struct SharedEdgeTraversal {
```

Comment:
Could it be useful to have the direction of this shared edge as part of the name? It's not immediately clear that it's entity -> ontology when reading the name :)

Reply:
Yes, that's a good point. I'm struggling to find a name, do you have a suggestion?

Reply:
Not sure other than `EntityToOntologySharedEdge` which is quite verbose, and potentially used somewhere else

Reply:
Sure that's fine with me!